CMakeLists.txt

```diff
@@ -260,6 +260,7 @@ target_precompile_headers(openttd_lib
     src/stdafx.h
     src/core/format.hpp
 )
+set_source_files_properties(src/3rdparty/fmt/format.cc PROPERTIES SKIP_PRECOMPILE_HEADERS ON)
```

Comment:
It seems we need to upgrade `set_compile_flags` and `process_compile_flags` to support more properties.
And I just noticed `set_compile_flags` is unused since https://github.com/OpenTTD/OpenTTD/commit/bba6ad1f4e3b676da881a05939f86fc6a70396fe.

Reply:
If `set_compile_flags` is unused, would it not make more sense to just remove it and `process_compile_flags` instead? It already seems needlessly complicated.